gprofiler/profilers/java.py

```diff
@@ -420,18 +424,20 @@ def _profile_ap_process(self, ap_proc: AsyncProfiledProcess) -> Optional[StackTo
                     f"async-profiler is still running in {ap_proc.process.pid}, even after trying to stop it!"
                 )
 
-        self._stop_event.wait(self._duration)
-
-        if ap_proc.process.is_running():
+        try:
+            wait_event(self._duration, self._stop_event, lambda: not ap_proc.process.is_running(), interval=1)
+        except TimeoutError:
+            # Process still running
             ap_proc.stop_async_profiler(True)
         else:
+            # Process terminated, was it due to an error?
+            hs_err_path = f'{ap_proc.host_cwd}/hs_err_pid{ap_proc.process.pid}.log'
```

Comment:
Parse commandline / env per the description in https://docs.oracle.com/javase/7/docs/webnotes/tsg/TSG-VM/html/felog.html and use that path if not default?

Reply:
Added.